### ISSUE TITLE
Integrate node-problem-detector with e2e test infrastructure

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kubernetes#30811]
Do the existing e2e tests fail if a node panicked during the test?

We are seeing issues like #30706, which makes me think we might not be detecting this.

It is actually surprisingly difficult to tell that a node has panicked with k8s, because k8s self-heals very quickly.  Also, I haven't yet figured out how to get a kernel panic into journald (on AWS), so the only place to see them is on the AWS console output.  But hopefully they would just be in the syslog on containervm.  A simple way to detect this is to look at the uptime of the nodes.

**Comment by justinsb** [comment]
cc @dchen1107

**Comment by dchen1107** [comment]
@justinsb We built node-problem-detector to make the kernel / filesystem issues visible. Also in our jenkins for aws, we eneabled node-problem-detector through https://github.com/kubernetes/test-infra/pull/392/files

And the support specifically for the kernel panic like #30706 is addressed by https://github.com/kubernetes/node-problem-detector/pull/22

**Comment by dchen1107** [comment]
@justinsb I re-title the issue to reflect what we really want here. 

@Random-Liu To finish this, we need

* Get https://github.com/kubernetes/test-infra/pull/392/files merged and build a new image for this daemonset
* Tried to fix / workaround https://github.com/kubernetes/kubernetes/issues/28687, so that  node-problem-detector daemonset will not be scheduled on our master node
* Updates e2e test infrastructure to always get node conditions and events, and logs those meaningful information. 
* Of course, a lot of existing tests might have to be changed since they assume some deterministic testing environment.

**Comment by dims** [comment]
switching this to 1.6 as it's too late for 1.5. ok? (please switch it right back if you disagree)

**Comment by dchen1107** [comment]
In 1.6 release, we have done the following:

1) deploy node-problem-detector daemonset with e2e test by default
2) collect node-problem-detector log for the debugging. 

I punt this to 1.7.

**Comment by Random-Liu** [comment]
Initially we planed to integrate node problem detector with the e2e framework to:
1) Detect node problems during e2e test.
2) Make sure NPD is actually working.

In 1.6 release, for 1) we collected NPD log, which is good enough for debugging https://github.com/kubernetes/kubernetes/pull/41949; 2) Add real case NPD e2e test to make sure it is working properly https://github.com/kubernetes/kubernetes/pull/42454.

I think it is safe to punt this to 1.7.
